test-d/get.ts

```diff
@@ -0,0 +1,49 @@
+import {Get} from '../ts41/get';
+import {expectTypeOf} from 'expect-type';
+
+interface ApiResponse {
+  hits: {
+    hits: Array<{
+      _id: string;
+      _source: {
+        name: Array<{
```

Comment:
```suggestion
        name: ReadonlyArray<{
```

just to ensure it works with readonly arrays too.

Reply:
Added tests for `readonly`, `ReadonlyArray`, as well as nullable/undefinedable props below.